cdc/processor/processor.go

```diff
@@ -407,53 +410,49 @@ func (p *processor) IsRemoveTableFinished(tableID model.TableID) (model.Ts, bool
 			zap.String("namespace", p.changefeedID.Namespace),
 			zap.String("changefeed", p.changefeedID.ID),
 			zap.Uint64("checkpointTs", tableCheckpointTs),
-			zap.Int64("tableID", tableID),
+			zap.Int64("tableID", span.TableID),
 			zap.Any("tableStatus", state))
 		return 0, false
 	}
 
 	if p.pullBasedSinking {
-		stats := p.sinkManager.GetTableStats(tableID)
-		p.sourceManager.RemoveTable(tableID)
-		p.sinkManager.RemoveTable(tableID)
+		stats := p.sinkManager.GetTableStats(span.TableID)
+		p.sourceManager.RemoveTable(span.TableID)
+		p.sinkManager.RemoveTable(span.TableID)
 		if p.redoManager.Enabled() {
-			p.redoManager.RemoveTable(tableID)
+			p.redoManager.RemoveTable(span.TableID)
 		}
 		log.Info("table removed",
 			zap.String("captureID", p.captureInfo.ID),
 			zap.String("namespace", p.changefeedID.Namespace),
 			zap.String("changefeed", p.changefeedID.ID),
-			zap.Int64("tableID", tableID),
+			zap.Int64("tableID", span.TableID),
 			zap.Uint64("checkpointTs", stats.CheckpointTs))
 
 		return stats.CheckpointTs, true
 	}
-	table := p.tables[tableID]
+	table := p.tables[span.TableID]
 	p.metricRemainKVEventGauge.Sub(float64(table.RemainEvents()))
 	table.Cancel()
 	table.Wait()
-	delete(p.tables, tableID)
+	delete(p.tables, span.TableID)
 
 	checkpointTs := table.CheckpointTs()
 	log.Info("table removed",
 		zap.String("captureID", p.captureInfo.ID),
 		zap.String("namespace", p.changefeedID.Namespace),
 		zap.String("changefeed", p.changefeedID.ID),
-		zap.Int64("tableID", tableID),
+		zap.Int64("tableID", span.TableID),
 		zap.Uint64("checkpointTs", checkpointTs))
 	return checkpointTs, true
 }
 
 // GetAllCurrentTables implements TableExecutor interface.
-func (p *processor) GetAllCurrentTables() []model.TableID {
+func (p *processor) GetAllCurrentTables() int {
```

Comment:
Is it possible that we care about all tableID ?

How about adding a new method `GetAllTableCount` here?

Reply:
I checked code base, no one needs tableID. I will rename the method.